src/main.ts

```diff
@@ -65,27 +72,31 @@ export async function run(inputs: RunInputs): Promise<void> {
     const outputPath = path.join(tempDir(), ATTESTATION_FILE_NAME)
     core.setOutput('bundle-path', outputPath)
 
-    const att = await createAttestation(subjects, predicate, {
+    const opts: CreateAttestationOptions = {
       sigstoreInstance,
       pushToRegistry: inputs.pushToRegistry,
       githubToken: inputs.githubToken
-    })
+    }
 
-    logAttestation(subjects, att, sigstoreInstance)
+    let atts: AttestResult[]
+    if (inputs.singleSubjectAttestations) {
+      atts = await createSingleSubjectAttestations(subjects, predicate, opts)
+    } else {
+      atts = await createMultiSubjectAttestation(subjects, predicate, opts)
+    }
```

Comment:
The core of `createAttestation` is split out into two functions. The first returns an array `atts` such that `len(atts) == len(subjects)`, and each `attestationSubjects` array has length 1. The second returns an array `atts` of length 1, with `atts[0].attestationSubjects == subjects`.